torch/csrc/jit/export.cpp

```diff
@@ -756,6 +757,174 @@ std::tuple<std::string, RawDataExportMap> ExportGraph(
   return std::make_tuple(model_proto.SerializeAsString(), graph_encoder.get_raw_data_export_map());
 }
 
+class PyTorchFileWriter {
+public:
+  PyTorchFileWriter(const std::string& filename){
+    fp = std::fopen(filename.c_str(), "wb");
+    writeFileHeader();
+  }
+
+  // Serialize a tensor to file, then return its offset
+  size_t serializeTensor(const std::string& name, at::Tensor t) {
```

Comment:
This should not take a name. It should take data and return an 8-byte key to where that data is. What the key is should be abstract in the API of this format.

Reply:
Signature should be:
```uint64_t writeData(char* data, size_t size_in_bytes)```

This should not have to know what a tensor is. We will use it to store non-tensor data.